backend/src/StdLibCliHost/Libs/Cli.fs

```diff
@@ -143,6 +145,118 @@ let fns : List<BuiltInFn> =
         | _ -> incorrectArgs ()
       sqlSpec = NotQueryable
       previewable = Impure
+      deprecated = NotDeprecated }
+
+
+    { name = fn [ "Cli" ] "executeFunction" 0
+      typeParams = []
+      parameters =
+        [ Param.make "functionName" TString ""
+          Param.make "args" (TList TString) "" ]
+      returnType =
+        TypeReference.result
+          TString
+          (TCustomType(FQName.BuiltIn(typ [ "Cli" ] "ExecutionError" 0), []))
+      description = "Executes an arbitrary Dark function"
+      fn =
+        function
+        | state, [], [ DString functionName; DList args ] ->
+          uply {
+            let err (msg : string) (metadata : List<string * string>) =
+              let metadata = metadata |> List.map (fun (k, v) -> k, DString v) |> Map
+              Dval.resultError (
+                Dval.record
+                  (FQName.BuiltIn(typ [ "Cli" ] "ExecutionError" 0))
+                  [ "msg", DString msg; "metadata", DDict metadata ]
+              )
+
+            let exnError (e : exn) : Dval =
+              let msg = Exception.getMessages e |> String.concat "\n"
+              let metadata =
+                Exception.toMetadata e |> List.map (fun (k, v) -> k, string v)
+              err msg metadata
+
+            try
+              let parts = functionName.Split('.') |> List.ofArray
+              let name = NEList.ofList "PACKAGE" parts
+              let resolver = LibParser.NameResolver.fromExecutionState state
+              let fnName =
+                LibParser.NameResolver.FnName.resolve
+                  resolver
+                  []
+                  (WT.Unresolved name)
+
+              match fnName, args with
+              | Ok fnName, firstArg :: additionalArgs ->
+                match
+                  NEList.find Dval.isFake (NEList.ofList firstArg additionalArgs)
+                with
+                | Some fakeArg -> return fakeArg
+                | None ->
+                  let desc = fnName |> PT2RT.FnName.toRT
+                  let! fn =
+                    match desc with
+                    | FQName.Package pkg ->
+                      uply {
+                        let! fn = state.packageManager.getFn pkg
+                        return Option.map packageFnToFn fn
+                      }
+
+                    | _ ->
+                      Exception.raiseInternal "Only package functions are executable from the CLI" [ "fn", fn ]
```

Comment:
(I just updated this error message) maybe this isn't the best error message actually - the issue here is that we just tried to create a Package fn name, and _somehow_ it's not